src/array_api_extra/_lib/_testing.py

```diff
@@ -47,43 +53,64 @@ def _check_ns_shape_dtype(
     msg = f"namespaces do not match: {actual_xp} != f{desired_xp}"
     assert actual_xp == desired_xp, msg
 
-    actual_shape = actual.shape
-    desired_shape = desired.shape
-    if is_dask_namespace(desired_xp):
-        # Dask uses nan instead of None for unknown shapes
-        if any(math.isnan(i) for i in cast(tuple[float, ...], actual_shape)):
-            actual_shape = actual.compute().shape  # type: ignore[attr-defined]  # pyright: ignore[reportAttributeAccessIssue]
-        if any(math.isnan(i) for i in cast(tuple[float, ...], desired_shape)):
-            desired_shape = desired.compute().shape  # type: ignore[attr-defined]  # pyright: ignore[reportAttributeAccessIssue]
-
-    msg = f"shapes do not match: {actual_shape} != f{desired_shape}"
-    assert actual_shape == desired_shape, msg
-
-    msg = f"dtypes do not match: {actual.dtype} != {desired.dtype}"
-    assert actual.dtype == desired.dtype, msg
+    if check_shape:
+        actual_shape = actual.shape
+        desired_shape = desired.shape
```

Comment:
This may fail if we start using it in scipy, because scipy overrides array_namespace to return numpy for scalars and lists. Maybe out of scope for this PR I though?